contracts/IAdapter.sol

```diff
@@ -7,39 +7,40 @@ interface IAdapter {
     // ethIndex is index of ETH amount in amounts array - 255 means no eth
     function invest(
         address onBehalfOf,
-        address[] calldata tokens, 
-        uint256[] calldata amounts, 
-        address to, 
+        address[] calldata tokens,
+        uint256[] calldata amounts,
+        address to,
         uint8 ethIndex
-    ) external returns (uint256[] memory amountsTransferred);
+    ) external payable returns (uint256[] memory amountsTransferred);
 
     // ethIndex is index of ETH amount in amounts array - 255 means no eth
     // Adapter needs to claim ither reward tokens as well
     function redeem(
         address onBehalfOf,
-        address[] calldata tokens, 
-        uint256[] calldata amounts, 
-        address to, 
+        address tokenToBurn,
+        address[] calldata tokens,
+        uint256[] calldata amounts,
+        address to,
         uint8 ethIndex
-    ) external returns (uint256[] memory amountsTransferred);
+    ) external payable returns (uint256[] memory amountsTransferred);
```

Comment:
I think this one does not need to be `payable`

Reply:
True, same for borrow